### PR TITLE
Improved descriptions of checks

### PR DESCRIPTION
**Pull request by FreHu**
- improved description texts, added more context and instructions if possible
- always use get_description method. It now takes a variable number of
strings to make this work

### CODE REVIEW

File: src/rules/if_in_if.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ export class IfInIf extends ABAPRule {
   }
 
   public getDescription(): string {
-    return "IF in IF";
+    return "IF in IF. Use IF cond1 AND cond2 instead. ";
```
Comment (larshp) [inline src/rules/if_in_if.ts]: there are more cases, but anyhow, this description is better

File: src/rules/no_public_attributes.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ import {Visibility} from "../abap/types";
 
 /** Checks that classes don't contain any public attributes */
 export class NoPublicAttributesConf extends BasicRuleConfig {
-// todo,  public allowReadOnly: boolean = false;
+  public allowReadOnly: boolean = false;
```
Comment (larshp) [inline src/rules/no_public_attributes.ts]: this not implemented yet, hence the commented code

File: src/rules/syntax/parser_error.ts [hunk 3]
```diff
@@ -21,8 +21,8 @@ export class ParserError extends ABAPRule {
     return "parser_error";
   }
 
-  public getDescription(): string {
-    return "Parser error(Unknown statement)";
+  public getDescription(statement: string, abapVersion: string): string {
+    return "Statament does not exist in ABAP" + abapVersion + ": '" + statement + "' If it does, this is a parser error.";
```
Comment (larshp) [inline src/rules/syntax/parser_error.ts]: typo, plus I think this description is too long to fit into a terminal(the filename is shown in the same line)

File: src/rules/use_new.ts [hunk 4]
```diff
@@ -19,7 +19,7 @@ export class UseNew extends ABAPRule {
   }
 
   public getDescription(): string {
-    return "Use NEW #( )";
+    return "Use NEW #( ) to initialize object.";
```
Comment (larshp) [inline src/rules/use_new.ts]: "instantiate" ?

File: src/rules/no_public_attributes.ts (hunk 2 above)
Comment (FreHu) [reply src/rules/no_public_attributes.ts]: I think I was just trying to figure out if it works and forgot to comment it back.

File: src/rules/use_new.ts (hunk 4 above)
Comment (FreHu) [reply src/rules/use_new.ts]: fixed

File: src/rules/syntax/parser_error.ts (hunk 3 above)
Comment (FreHu) [reply src/rules/syntax/parser_error.ts]: shortened

File: src/rules/if_in_if.ts (hunk 1 above)
Comment (FreHu) [reply src/rules/if_in_if.ts]: keeping it as is then

**Comment by larshp** [comment]
argh, should have merged before releasing 2.0.2, but anyhow, next release is probably later today or in the weekend 🚢

**Pull request merged**